client/auth.go

```diff
@@ -60,7 +60,7 @@ func (client *Client) InjectAuthenticationHeader(req *http.Request, path string)
 			Value: client.AuthToken.Token,
 		})
 		return req, nil
-	} else if client.privatekey != "" && client.adminCert != "" {
+	} else if client.privatekey != "" && (client.adminCert != "" || client.appUserName != "") {
```

Comment:
this doesn't look correct.  I know you're trying to pass in the appUserName twice, but why not pass i nthe adminCert name instead.  If you pass in the adminCertName, then you dont need this change here.

Reply:
Removed appusername here.